### PR TITLE
Describe proposed interface

### PR DESCRIPTION
**Pull request by jessemyers**
Update README to document proposed `nodule-config` API

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,105 +2,137 @@
...
+            testing: true,
+            // factories may have different behavior during local development
+            debug: false,
+        };
+
+With this `Metadata`, `Nodule` can load configuration using loader functions:
+
+        import { Nodule, loadFromEnvironment } from 'nodule-config';
+
+        const nodule = new Nodule(metadata);
```
Comment (cdm-ium) [inline README.md]: If nodule is like what our object graph is, the constructor here would take `metadata` and the `loader(s)` and build config from them.

Not opposed to exposing a function for loading config from arbitrary constructors after the nodule is constructed, but curious why you want to be able to pass the loaders in dynamically vs when creating a `Nodule`.

File: README.md [hunk 2]
```diff
@@ -2,105 +2,137 @@
...
+        nodule.loader(loadFromEnvironment).then(bottle => myInitFunc(bottle));
+
+Common loader functions also have shortcuts:
+
+        nodule.loadFromEnvironment().then(bottle => myInitFunc(bottle));
```
Comment (cdm-ium) [inline README.md]: I ended up putting the `loadFromEnvironment` and other loaders outside the nodule as I think it makes sense for `Nodule` to expect a common api from loaders and let people pass in whatever loaders they want. I think haven't default loaders is a good idea, but not sure about exposing them on the class itself via class functions.

File: README.md [hunk 3]
```diff
@@ -2,105 +2,137 @@
...
+    import { bind } from 'nodule-config';
+
+    bind('foo', myFactoryFunc);
+
+Configuration should typically have sane defaults (esp. for local development). `Nodule` enables configuration of
+such defaults:
+
+    import { setDefaults } from 'nodule-config';
+
+    setDefaults('foo', {
```
Comment (cdm-ium) [inline README.md]: 👍

File: README.md (hunk 2 above)
Comment (jessemyers) [reply README.md]: I don't know what you mean by "exposing them on the class itself" vs "class functions".

**Pull request merged**